tests/Detector.Tests/Go/GoDetectorTest.cs

```diff
@@ -0,0 +1,121 @@
+﻿// --------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+// --------------------------------------------------------------------------------------------
+
+using System;
+using System.IO;
+using Microsoft.Extensions.Logging.Abstractions;
+using Microsoft.Extensions.Options;
+using Microsoft.Oryx.Detector.Go;
+using Microsoft.Oryx.Tests.Common;
+using Xunit;
+namespace Microsoft.Oryx.Detector.Tests.Go
+{
+    public class GoDetectorTest : IClassFixture<TestTempDirTestFixture>
+    {
+
+        private const string GoFileWithVersion = @"
```

Comment:
nit: GolangModFileWithMajorMinorVersion

Reply:
add 2 more 
GolangModFileWithOnlyMajorVersion
GolangModFileWithMajorMinorPatchVersion

Reply:
Add additional tests to PR